apollo-router/src/query_planner/bridge_query_planner.rs

```diff
@@ -41,6 +42,11 @@ impl BridgeQueryPlanner {
         introspection: Option<Arc<Introspection>>,
         configuration: Arc<Configuration>,
     ) -> Result<Self, QueryPlannerError> {
+        // The variables deduplication parameter lives in the traffic_shaping section of the config
+        let deduplicate_variables = configuration
+            .plugin_configuration("apollo.traffic_shaping")
```

Comment:
could this be a (static) method of the traffic shaping plugin? To notice it easily if something changes in that plugin

Reply:
hmmm not really sure how, but there's probably a way for me to ask traffic_shaping to give me its settings once i ve passed a configuration in Oo

Reply:
Should this config really be in `traffic_shaping`?

Reply:
something simple like:
```rust
impl TrafficShaping {
  fn get_configuration_deduplicate_variables(configuration: &Configuration) -> bool {
     configuration
            .plugin_configuration("apollo.traffic_shaping").map(|conf| conf.get("deduplicate_variables") == Some(&serde_json::Value::Bool(true)))
            .unwrap_or_default()
  }
}
```

Just to have the code related to this part of the configuration in the plugin so we don't miss something if config options change

Reply:
> Should this config really be in traffic_shaping?

I don't think so, while its goal is to reduce the size of subgraph requests and responses, it's more related to query plan execution

Reply:
I see.

RE: traffic_shaping yml config: I think we should revisit this as part of #1500

Reply:
fixed in 5b5612afa0b20fbec3b64f77234a616512e136cb